packages/syft/src/syft/service/network/node_peer.py

```diff
@@ -42,12 +44,55 @@ class NodePeer(SyftObject):
 
     def update_routes(self, new_routes: List[NodeRoute]) -> None:
         add_routes = []
-        existing_routes = set(self.node_routes)
+        new_routes: List[NodeRoute] = self.update_route_priorities(new_routes)
         for new_route in new_routes:
-            if new_route not in existing_routes:
+            existed, index = self.existed_route(new_route)
+            if not existed:
                 add_routes.append(new_route)
+            else:
+                # if the route already exists, we do not append it to self.new_route,
+                # but update its priority
+                self.node_routes[index].priority = new_route.priority
+
         self.node_routes += add_routes
 
+    def update_route_priorities(self, new_routes: List[NodeRoute]) -> List[NodeRoute]:
+        """
+        Since we pick the newest route has the highest priority, we
+        update the priority of the newly added routes here to be increments of
+        current routes' highest priority.
+        """
+        current_max_priority = max(route.priority for route in self.node_routes)
+        for route in new_routes:
+            route.priority = current_max_priority + 1
+            current_max_priority += 1
+        return new_routes
+
+    def existed_route(self, route: NodeRoute) -> Tuple[bool, int]:
```

Comment:
This function could just return an `Optional[int]`?

Reply:
thanks @kiendang